sway/realtime.c

```diff
@@ -0,0 +1,40 @@
+#include <sys/resource.h>
+#include <sched.h>
+#include <unistd.h>
+#include <pthread.h>
+#include "sway/server.h"
+#include "log.h"
+
+static void child_fork_callback(void) {
+	struct sched_param param;
+
+	param.sched_priority = 0;
+
+	int ret = pthread_setschedparam(pthread_self(), SCHED_OTHER, &param);
+	if (ret != 0) {
+		sway_log(SWAY_ERROR, "Failed to reset scheduler policy on fork");
+	}
+}
+
+void gain_realtime(void) {
+	int prio = sched_get_priority_min(SCHED_RR);
```

Comment:
Hmm, so pipewire ends up having priority 9, but `sched_get_priority_min(SCHED_RR)` is -2 so we're below pipewire which is a little annoying. I wonder where the pipewire's priority comes from - rtkit seems to ask for priority 0?

Reply:
Pipewire uses the `MaxRealtimePriority`property from RTKit  to request the highest possible priority from RTKit. This sounds fine and also sway requesting the lowest sounds good. According to `man 2 sched_get_priority_min`:
```
Processes  with  numerically  higher  priority  values are scheduled before processes with numerically lower priority values. Thus, the value returned by sched_get_priority_max() will be greater than the value returned by sched_get_priority_min().
```
So pipewire should be scheduled before sway.

Reply:
> Pipewire uses the `MaxRealtimePriority` property from RTKit to request the highest possible priority from RTKit.

Oh, I just realized that I was checking prio, not rtprio. That explains why I was seeing weird numbers. D'oh.

> So pipewire should be scheduled before sway.

Yeah, my thought was that it makes sense that we as the graphical window system would have either same or higher priority than audio and screen capture routing. It does seem a bit aggressive from pipewire's perspective to ask for the highest priority, outranking even a bunch of kernel threads...

(This is of course nowhere near as important as just being SCHED_RR in the first place - even if our priority is wrong, we're competing with a much smaller group of processes.)